packages/flutter/test/cupertino/nav_bar_test.dart

```diff
@@ -2087,6 +2087,212 @@ void main() {
+    expect(renderOpacity?.opacity.value, 0.0);
+
+    // Scroll to just past the halfway point of the large title.
+    final TestGesture scrollGesture2 = await tester.startGesture(tester.getCenter(find.byType(Scrollable)));
+    await scrollGesture2.moveBy(const Offset(0.0, -(largeTitleHeight / 2) - 1));
+    await scrollGesture2.up();
+    await tester.pumpAndSettle();
+
+    // Expect the large title to snap up to the persistent nav bar.
+    expect(scrollController.position.pixels, bottomHeight + largeTitleHeight);
+    expect(renderOpacity?.opacity.value, 1.0);
+  });
+
+  testWidgets('Large title and bottom snap down when partially scrolled halfway up or less in automatic mode', (WidgetTester tester) async {
+    final ScrollController scrollController = ScrollController();
+    const double largeTitleHeight = 52.0;
+    const double bottomHeight = 100.0;
+
+    await tester.pumpWidget(
+      MediaQuery(
+        data: const MediaQueryData(),
+        child: CupertinoApp(
+          home: CustomScrollView(
+            controller: scrollController,
+            slivers: const <Widget>[
+              CupertinoSliverNavigationBar(
+                largeTitle: Text('Large title'),
+                middle: Text('middle'),
+                alwaysShowMiddle: false,
+                bottom: PreferredSize(
+                  preferredSize: Size.fromHeight(bottomHeight),
+                  child: Placeholder(),
+                ),
+                bottomMode: NavigationBarBottomMode.automatic,
+              ),
+              SliverFillRemaining(
+                child: SizedBox(
+                  height: 1000.0,
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+
+    final RenderAnimatedOpacity? renderOpacity = tester.element(find.text('middle')).findAncestorRenderObjectOfType<RenderAnimatedOpacity>();
+
+    expect(renderOpacity?.opacity.value, 0.0);
+    expect(scrollController.offset, 0.0);
+
+    // Scroll to the halfway point of the bottom widget.
+    final TestGesture scrollGesture1 = await tester.startGesture(tester.getCenter(find.byType(Scrollable)));
+    await scrollGesture1.moveBy(const Offset(0.0, -bottomHeight / 2));
+    await scrollGesture1.up();
+    await tester.pumpAndSettle();
+
+    // Expect the bottom to snap back to its extended height.
+    expect(scrollController.position.pixels, 0.0);
+    expect(renderOpacity?.opacity.value, 0.0);
+
+    // Scroll to the halfway point of the large title.
+    final TestGesture scrollGesture2 = await tester.startGesture(tester.getCenter(find.byType(Scrollable)));
+    await scrollGesture2.moveBy(const Offset(0.0, -(bottomHeight + largeTitleHeight / 2)));
+    await scrollGesture2.up();
+    await tester.pumpAndSettle();
+
+    // Expect the large title to snap back to its extended height, which is the
+    // same scroll offset as the fully-shrunk bottom widget.
+    expect(scrollController.position.pixels, bottomHeight);
```

Comment:
Same here. Should add a more direct check that the bottom widget is not visible.